Source/Endpoint.swift

```diff
@@ -74,16 +75,25 @@ extension Endpoint {
 
         return parameterEncoding.toAlamofire.encode(request, parameters: parameters).0
     }
+
+    // let's return a mutable request, that could be modify before it's be performed
+    func toNSMutableURLRequest() -> NSMutableURLRequest {
```

Comment:
the Moya current flow is `Endpoint` -> `NSURLRequest` (by `RequestClosure` definition) then `NSURLRequest` -> `AlamoRequest` (by [sendRequest](https://github.com/Moya/Moya/blob/master/Source/Moya.swift#L166))

`toNSMutableURLRequest` is used in `RequestClosure` [for example](https://github.com/Moya/Moya/blob/master/docs/Endpoints.md#request-mapping) 

I just keeping the behavior, and I think that's fine, I've read Alamofire's Request, most except authenticate are no use on preparation (authenticate can be done by plugin), and that's help with decoupling with Alamofire (although it's no necessary)